apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -80,61 +102,74 @@ export class MessageMatcher {
   }
 
-  private splitToSyncAsync(filter: StepFilter) {
-    const asyncFilters = findFilterPartByType(filter.children, 'webhook') as IWebhookFilterPart[];
-    const syncFilters = findFilterPartByType(filter.children, ['payload', 'subscriber']) as IFieldFilterPart[];
+  private splitFilters(filter: StepFilter) {
+    const webhookFilters = filter.children.filter((childFilter) => childFilter.on === 'webhook');
 
-    return { asyncFilters, syncFilters };
+    const otherFilters = filter.children.filter((childFilter) => childFilter.on !== 'webhook');
+
+    return { webhookFilters, otherFilters };
   }
 
   private async handleAndFilters(
     filter: StepFilter,
     variables: IFilterVariables,
-    command: SendMessageCommand
+    command: SendMessageCommand,
+    filterProcessingDetails: FilterProcessingDetails
   ): Promise<boolean> {
-    const { asyncFilters, syncFilters } = this.splitToSyncAsync(filter);
+    const { webhookFilters, otherFilters } = this.splitFilters(filter);
 
-    const foundSyncFilterMatches = syncFilters.filter((i) => this.processFilterEquality(variables, i));
-    if (syncFilters.length !== foundSyncFilterMatches.length) {
+    const matchedOtherFilters = await filterAsync(otherFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    );
+    if (otherFilters.length !== matchedOtherFilters.length) {
       return false;
     }
 
-    const foundAsyncFilterMatches = await filterAsync(asyncFilters, (i) => this.processFilter(variables, i, command));
+    const matchedWebhookFilters = await filterAsync(webhookFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    );
 
-    return foundAsyncFilterMatches.length === asyncFilters.length;
+    return matchedWebhookFilters.length === webhookFilters.length;
   }
 
   private async handleOrFilters(
     filter: StepFilter,
     variables: IFilterVariables,
-    command: SendMessageCommand
+    command: SendMessageCommand,
+    filterProcessingDetails: FilterProcessingDetails
   ): Promise<boolean> {
-    const { asyncFilters, syncFilters } = this.splitToSyncAsync(filter);
+    const { webhookFilters, otherFilters } = this.splitFilters(filter);
 
-    const syncRes = syncFilters.find((i) => this.processFilterEquality(variables, i));
-    if (syncRes) {
+    const foundFilter = await findAsync(otherFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    );
+    if (foundFilter) {
       return true;
     }
 
-    return !!(await findAsync(asyncFilters, (i) => this.processFilter(variables, i, command)));
+    return !!(await findAsync(webhookFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    ));
   }
 
   private async processIsOnline(
     filter: IRealtimeOnlineFilterPart | IOnlineInLastFilterPart,
-    command: SendMessageCommand
+    command: SendMessageCommand,
+    filterProcessingDetails: FilterProcessingDetails
```

Comment:
in the `processIsOnline` function we are calling `filterProcessingDetails.addCondition` to add the condition details, like the information about the filter `on` field, field, expected, actual, operator, passed etc.